analysis/quantifier-comparisons/pre-processing-benchmarking-metrics.Rmd

```diff
@@ -0,0 +1,612 @@
+
+
+```{r}
+# spread table for comparisons
+feature_compare <- feature_qc %>%
+  dplyr::filter(gene_id %in% detected_features) %>%
+  # spread the mean expression stats to one column per caller
+  tidyr::pivot_wider(id_cols = c(gene_id, sample),
+                     names_from = tool,
+                     values_from = mean) %>%
+  # drop rows with NA values to ease correlation calculations
+  tidyr::drop_na()
+```
+
+```{r}
+feature_compare %>% 
+  dplyr::group_by(sample) %>%
+  dplyr::summarize(
+    cr_al_cor = cor(cellranger, alevin, method = "spearman"),
+    cr_al_fry_cor =cor(cellranger, alevin_fry, method = "spearman"),
+    cr_al_fry_sketch_cor = cor(cellranger, alevin_fry_sketch, method = "spearman"),
+    cr_ka_cor = cor(cellranger, kallisto, method = "spearman")
+  )
+```
+So here we have correlation of mean expression/gene and the correlations here are much more varied than when looking at UMI/cell. We see that the highest correlation is between Alevin and Alevin-fry (without sketch), while Kallisto has the lowest correlation. It appears that the `--sketch` mode, although increases time (shown elsewhere), doesn't appear to have an increased advantage over Alevin-fry in terms of quantification. 
+
+```{r}
+ggplot(feature_compare, aes(x = cellranger, y = alevin_fry)) +
+  geom_point(size = 0.5, alpha = 0.1) + 
+  facet_wrap(~ sample) + 
+  scale_x_log10() + 
+  scale_y_log10() + 
+  labs(x = "Cell Ranger Mean Gene Expression", y = "Alevin-fry Mean Gene Expression") + 
+  theme_classic()
+```
+```{r}
+ggplot(feature_compare, aes(x = cellranger, y = kallisto)) +
+  geom_point(size = 0.5, alpha = 0.1) + 
+  facet_wrap(~ sample) + 
+  scale_x_log10() + 
+  scale_y_log10() + 
+  labs(x = "Cell Ranger Mean Gene Expression", y = "Kallisto Mean Gene Expression") + 
+  theme_classic()
+```
+Interestingly, Alevin-fry has a higher correlation than Kallisto, the pattern is much different. The mean gene expression tends to be higher in genes in Alevin-fry than in Cellranger. In kallisto, there are genes that are higher in both kallisto and cellranger. 
+
+In conclusion, it appears that there is some variability between the tools in cell barcode assignment, with cellranger assigning what appears to be more cells, but a lower number of UMI/cells and gene/cells in compared to Kallisto and Alevin-fry. For the most part, there is high correlation of each tool with cellranger, but Alevin-Fry/Alevin have the highest and Kallisto the lowest. Although Kallisto has the fastest compute time, Alevin-Fry is still much faster than cellranger and uses a fraction of the memory while obtaining similar quantification information. The biggest discrepancy is in the number of cells that are identified, so perhaps it is worth looking into the other resolution methods used in Alevin-fry, as we are using the "full" resolution method here. 
```

Comment:
I would phrase this probably more as a difference in cell filtering; I expect that cellranger is being more permissive. So cellranger is filtering fewer cells out, not that the barcode assignment is necessarily different.

I don't think the resolution methods for alevin-fry are at the cell level, but rather the read level, as to how a read gets counted if it maps to more than one gene/transcript.